### ISSUE TITLE
Restore automated tests

### ISSUE THREAD

**Issue by tuliomir** [HathorNetwork/hathor-wallet#696]
Automated tests were implemented through #357 and #365 , but disabled on #416 .

We should reactivate those tests to improve our QA process, both unit and e2e. It would be nice to have visual regression tests and simulated ledger interactions as well.

# Found issues
The unit tests are not able to run because Jest cannot process the modern code we're using. This happens because we're using an older version of jest that is defined by CRA, and we have to decide an upgrade path ( mapped in #441 ) before we can upgrade jest.

**Comment by tuliomir** [comment]
This task was partially implemented by #698, #736, and will be handled from now on by #704
